src/cls/rbd/cls_rbd.cc

```diff
@@ -7492,6 +7630,10 @@ int group_snap_remove(cls_method_context_t hctx,
 
   CLS_LOG(20, "removing snapshot with key %s", snap_key.c_str());
   int r = cls_cxx_map_remove_key(hctx, snap_key);
+  if (r == 0) {
+    std::string snap_order_key = group::snap_order_key(snap_id);
+    cls_cxx_map_remove_key(hctx, snap_order_key);
```

Comment:
> I guess `ENOENT` would be expected for older snapshots

This was missed -- the updated method should be able to remove older snapshots without error.

Reply:
It would be good to add `group_snap_remove_no_order` test for this too: create a snapshot, remove its order key, assert that `group_snap_remove()` succeeds.